packages/react-scripts/scripts/utils/verifyTypeScriptSetup.js

```diff
@@ -218,6 +218,14 @@ function verifyTypeScriptSetup() {
   if (appTsConfig.compilerOptions == null) {
     appTsConfig.compilerOptions = {};
     firstTimeSetup = true;
+  } else {
+    // This is bug fix code of https://github.com/facebook/create-react-app/issues/9868
+    // Bellow code release variable from non-extensible and freeze status.
+    appTsConfig.compilerOptions = { ...appTsConfig.compilerOptions };
```

Comment:
making sure that all properties are not frozen by cloning the whole object:
```suggestion
    appTsConfig.compilerOptions = JSON.parse(JSON.stringify(appTsConfig.compilerOptions));
```

Reply:
If it's only data, it should work. But if there are functions with references inside, they won't work afterwards. But since this should simply be parsed data from tsconfig.json I suggest that it'll work fine.

But I'd say that this change should have much earlier in the code. At line 185 is a comment `// Calling this function also mutates the tsconfig above,`. Though I'd put the change here and reuse this comment to explain why the object needs to be cloned in this place.

Reply:
IMO the safest way to deal with such issues is to refactor this function to not mutate any of those configs, but that would require a lot of re-writing, so I guess that's not optimal too.